app/src/main/java/org/simple/clinic/user/UserSession.kt

```diff
@@ -183,19 +183,24 @@ class UserSession @Inject constructor(
   }
 
   fun refreshLoggedInUser(): Completable {
-    return loggedInUser()
+    return requireLoggedInUser()
         .firstOrError()
-        .map { (user) ->
-          if (user == null) {
-            throw AssertionError("User isn't logged in yet.")
-          }
-          user
-        }
-        .flatMap { registrationApi.findUser(it.phoneNumber) }
-        .flatMapCompletable { userPayload ->
-          val user = userFromPayload(userPayload, User.LoggedInStatus.LOGGED_IN)
-          val userFacilities = userPayload.facilityUuids
-          storeUser(user, userFacilities)
+        .flatMapCompletable { loggedInUser ->
+          registrationApi.findUser(loggedInUser.phoneNumber)
+              .flatMapCompletable { userPayload ->
+                // FIXME: This is a hack to handle the case where the user logged in status will
+                // not get set to LOGGED_IN when the PIN reset request is approved.
+                val finalLoggedInStatus = if (userPayload.status == UserStatus.APPROVED_FOR_SYNCING) {
```

Comment:
Earlier, when we used to query the user status in the home screen (to check whether the user has been approved from the backend for syncing), we used to hardcode the local `loggedInStatus` of the user to `LOGGED_IN` since that is what we used for a newly registered user since we only cared about the change in the `status` field.

However, when it comes to PIN reset, we set the local state to `PIN_RESET_REQUESTED` and this state must be maintained for necessary messaging in the UI. So this retains the previous `loggedInStatus` of the user we are refreshing (`LOGGED_IN` for a newly registered user, and `PIN_RESET_REQUESTED` for a user who's resetting the PIN) *until* the server says they have been approved, at which point we change the local `loggedInStatus` to `LOGGED_IN` again.